armi/physics/neutronics/crossSectionGroupManager.py

```diff
@@ -238,6 +238,23 @@ def getCandidateBlocks(self):
         """
         return [b for b in self if b.hasFlags(self._validRepresentativeBlockTypes)]
 
+    def _calcWeightedBurnup(self):
+        """
+        For a blockCollection that represents fuel, calculate the weighted average burnup
+
+        Notes
+        -----
+        - Only used for logging purposes
+        - Burnup needs to be weighted by heavy metal mass instead of volume
+        """
+        weightedBurnup = 0.0
+        totalWeight = 0.0
+        for b in self:
+            weighting = b.p.massHmBOL * self.getWeight(b) / b.getVolume()
```

Comment:
why divide by volume here?

Reply:
Yes, `getWeight` multiplies the weighting by volume. I wanted to just multiply the mass by the XSGM weighting factor.

Reply:
sorry, i was suggesting comment. if you prefer doc string thats fine but added suggestion